taraxa/state/dpos/precompiled/dpos_contract.go

```diff
@@ -468,53 +471,96 @@ func (self *Contract) Run(ctx vm.CallFrame, evm *vm.EVM) ([]byte, error) {
+// // + first we calculate theoretical maxamount of bonus votes `maxBonusVotes`
+// // + then `authorRewardRatio` is calculate included votes : to `maxBonusVotes`
+// // + then based on this ratio, author will receive propotion of reward from bonus votes reward
+// // + final reward can be tuned by `MaxBlockAuthorReward` that represents % of calculated bonus votes
+// - Vote reward is reduced by author reward
+// - Then for each validator vote and transaction propotion rewards are calculated and distributed 
+
+func (self *Contract) DistributeRewards(BlockAuthorAddr *common.Address, rewardsStats *rewards_stats.RewardsStats, feesRewards *FeesRewards) {
 	// When calling DistributeRewards, internal structures must be always initialized
 	self.lazy_init()
 
 	// Calculates number of tokens to be generated as block reward
 	blockReward := bigutil.Mul(self.amount_delegated, self.yield_percentage)
 	blockReward = bigutil.Div(blockReward, bigutil.Mul(Big100, self.blocks_per_year))
 
+	votesReward := bigutil.Big0
+	authorReward := bigutil.Big0
+	trxsReward := blockReward
+	// We need to handle case for block 1
+	if rewardsStats.TotalVotesWeight > 0 {
+		// Calculate propotion between votes and transactions
+		trxsReward = bigutil.Div(bigutil.Mul(blockReward, VotesToTrasnactionsRatio), Big100)
+		votesReward = bigutil.Sub(blockReward, trxsReward)
+
+		// In case block author included more than minimum required number of votes
+		if rewardsStats.BonusVotesWeight > 0 && self.cfg.MaxBlockAuthorReward > 0 {
+			maxBonusVotes := rewardsStats.MaxVotesWeight / 3
+			authorRewardRatio := rewardsStats.BonusVotesWeight * uint64(self.cfg.MaxBlockAuthorReward) / maxBonusVotes
+			validatorVoteReward := bigutil.Mul(big.NewInt(int64(rewardsStats.BonusVotesWeight)), votesReward)
+			validatorVoteReward = bigutil.Div(validatorVoteReward, big.NewInt(int64(rewardsStats.TotalVotesWeight)))
+			authorReward = bigutil.Div(bigutil.Mul(validatorVoteReward, big.NewInt(int64(authorRewardRatio))), Big100)
+			votesReward = bigutil.Sub(votesReward, authorReward)
+		}
+	}
+
 	totalUniqueTxsCountCheck := uint32(0)
+	totalVoteWeightCheck := uint64(0)
+	totalRewardCheck := bigutil.Big0
+	// Add reward to the author for additional included votes
+	if authorReward.Cmp(bigutil.Big0) == 1 {
+		block_author := self.validators.GetValidator(BlockAuthorAddr)
+		if block_author == nil {
+			panic("DistributeRewards - non existent block author")
+		}
+
+		validatorCommission := bigutil.Div(bigutil.Mul(authorReward, big.NewInt(int64(block_author.Commission))), Big10000)
+		delegatorsRewards := bigutil.Sub(authorReward, validatorCommission)
+
+		block_author.CommissionRewardsPool = bigutil.Add(block_author.CommissionRewardsPool, validatorCommission)
+		block_author.RewardsPool = bigutil.Add(block_author.RewardsPool, delegatorsRewards)
+
+		totalRewardCheck = bigutil.Add(totalRewardCheck, authorReward)
+		self.validators.ModifyValidator(BlockAuthorAddr, block_author)
```

Comment:
I know we dont really have unified code style in this contract(and go in general), but at least in this code block everything is camelcase except block_author

Reply:
I change it so block_author because you wanted it like that few comments above